File: all-submissions/0128-longest-consecutive-sequence/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums == null || nums.length ==0) return 0;
        if(nums.length ==1) return 1;
        Set<Integer> set = new HashSet<>();
        int smallest = Integer.MAX_VALUE;
        int largest = Integer.MIN_VALUE;
        int answer = 1;
        for(int num: nums){
            smallest = Math.min(smallest, num);
            largest = Math.max(largest, num);
            set.add(num);
        }
        for(int num: set){
            if(!set.contains(num-1)){
                int currentNum = num;
                int currConsecutive = 0;
                while(set.contains(currentNum)){
                    currConsecutive++;
                    currentNum++;
                }
                answer = Math.max(answer, currConsecutive);
            }
        }
        return answer;
    }
}

File: all-submissions/0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        longest_streak = 0
        visited = set()
        for num in nums_set:
            if num in visited:
                continue
            curr_streak = 1
            while num+1 in nums_set:
                visited.add(num)
                num+=1
                curr_streak +=1
            visited.add(num)
            longest_streak = max(longest_streak, curr_streak)
        return longest_streak
